src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/PreferLengthCountIsEmptyOverAnyAnalyzer.cs

```diff
@@ -0,0 +1,140 @@
+            "Performance",
+            RuleLevel.IdeSuggestion,
+            CreateLocalizableResourceString(nameof(PreferCountOverAnyDescription)),
+            false,
+            false
+        );
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.RegisterOperationAction(OnInvocationAnalysis, OperationKind.Invocation);
+        }
+
+        private static void OnInvocationAnalysis(OperationAnalysisContext context)
+        {
+            var invocation = (IInvocationOperation)context.Operation;
+            var originalMethod = invocation.TargetMethod.OriginalDefinition;
+            if (IsEligibleAnyMethod(originalMethod))
+            {
+                // The collection will be passed as the instance in VB and as an argument in C#. 
+                var firstArgument = invocation.Instance ?? invocation.Arguments[0].Value;
+                var type = (firstArgument as IConversionOperation)?.Operand.Type ?? firstArgument.Type;
+                if (HasEligibleIsEmptyProperty(type))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(IsEmptyDescriptor, invocation.Syntax.GetLocation()));
+
+                    return;
+                }
+
+                if (HasEligibleLengthProperty(type))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(LengthDescriptor, invocation.Syntax.GetLocation()));
+
+                    return;
+                }
+
+                if (HasEligibleCountProperty(type))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(CountDescriptor, invocation.Syntax.GetLocation()));
+
+                    return;
+                }
+            }
+        }
+
+        private static bool IsEligibleAnyMethod(IMethodSymbol method)
+        {
+            return method.Name == AnyText
+                   && method.ReturnType.SpecialType == SpecialType.System_Boolean
+                   && method.Language == LanguageNames.CSharp && method.Parameters.Length == 1
+                   || (method.Language == LanguageNames.VisualBasic && method.Parameters.Length == 0);
+        }
+
+        private static bool HasEligibleIsEmptyProperty(ITypeSymbol typeSymbol)
+        {
+            return typeSymbol.GetMembers(IsEmptyText)
+                .Any(symbol => symbol is IPropertySymbol property && property.Type.SpecialType == SpecialType.System_Boolean);
+        }
+
+        private static bool HasEligibleLengthProperty(ITypeSymbol typeSymbol)
+        {
+            if (typeSymbol is IArrayTypeSymbol)
+            {
+                return true;
+            }
+
+            return typeSymbol.GetMembers(LengthText)
+                .Any(symbol => symbol is IPropertySymbol property && property.Type.SpecialType == SpecialType.System_Int32);
```

Comment:
`uint` is common for Length/Count in UWP, e.g:

https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.data.itemindexrange.length?view=winrt-18362

I'm not sure if we should handle it too.

@stephentoub What do you think?

Reply:
Ping @stephentoub @buyaa-n for this question.

Reply:
Makes sense to cover